internal/cloud/state.go

```diff
@@ -69,6 +69,9 @@ type State struct {
 	// not effect final snapshots after an operation, which will always
 	// be written to the remote API.
 	stateSnapshotInterval time.Duration
+	// If the header, X-Terraform-Snapshot-Interval is not present then
+	// we will disable snapshots
+	disableIntermediateSnapshots bool
```

Comment:
Given the requirement that this only send intermediate snapshots when the remote API explicitly opts in to supporting them, I was expecting this to be `enableIntermediateSnapshots` defaulting to `false` and then get set to `true` only in `readSnapshotIntervalHeader`.

Given that in practice we should always call `RefreshState` before any calls to `ShouldPersistIntermediateState` it doesn't technically matter how we orient it, but I think making it start disabled and then get enabled better reflects how you described the underlying requirement and so I expect it will seem clearer to future maintainers.